rest/blip_sync.go

```diff
@@ -737,6 +740,33 @@ func (bh *blipHandler) handleRev(rq *blip.Message) error {
 		return base.HTTPErrorf(http.StatusBadRequest, "Missing docID or revID")
 	}
 
+	if deltaSrcRevID, isDelta := revMessage.deltaSrc(); isDelta {
+		bh.Logf(base.LevelDebug, base.KeySync, "docID: %s - Received rev message as delta", base.UD(docID))
+		if !bh.sgCanUseDeltas {
+			return base.HTTPErrorf(http.StatusBadRequest, "Deltas are disabled for this peer")
+		}
+
+		delta, err := rq.Body()
+		if err != nil {
+			return base.HTTPErrorf(http.StatusBadRequest, "Error getting delta body: %s", err)
+		}
+
+		deltaSrcBody, err := bh.db.GetRev(docID, deltaSrcRevID, true, nil)
+		if err != nil {
+			return base.HTTPErrorf(http.StatusNotFound, "Can't fetch doc for deltaSrc=%s %v", deltaSrcRevID, err)
+		}
+
+		// TODO: need to add deep-copy support to revcache so deltaSrcBody doesn't get mutated
```

Comment:
For this PR, make a deep copy here.  Filed separate follow-on enhancement to move that down to the rev cache (file a tracking ticket for the 'copy level' parameter on GetRev) (CBG-159)